lib/node.ts

```diff
@@ -1,10 +1,17 @@
 import AlgoliaAnalytics from "./insights";
 import { getFunctionalInterface } from "./_getFunctionalInterface";
+import { getRequesterForBrowser } from "./utils/getRequesterForBrowser";
 import { getRequesterForNode } from "./utils/getRequesterForNode";
 import { RequestFnType } from "./utils/request";
 
+export { getRequesterForBrowser, getRequesterForNode };
+
 export function createInsightsClient(requestFn: RequestFnType) {
   return getFunctionalInterface(new AlgoliaAnalytics({ requestFn }));
 }
 
-export default createInsightsClient(getRequesterForNode());
+export default createInsightsClient(
+  typeof window !== "undefined"
+    ? getRequesterForBrowser()
+    : getRequesterForNode()
+);
```

Comment:
you should add a `browser` key to the package.json instead of doing this kind of conditional

Reply:
If we choose to do that, users will get to use umd build instead of cjs, even when it's loaded via bundlers. What do you think?

Reply:
you can make a file like this for browser, no?

Reply:
ah, so you're saying like this?

- umd
- cjs for node
- cjs for browser